meteor/server/api/playout/syncFunction.ts

```diff
@@ -0,0 +1,201 @@
+import { Meteor } from 'meteor/meteor'
+import { ReadonlyDeep } from 'type-fest'
+import { MethodContext } from '../../../lib/api/methods'
+import { RundownPlaylistId, RundownPlaylist, RundownPlaylists } from '../../../lib/collections/RundownPlaylists'
+import { waitForPromise } from '../../../lib/lib'
+import { ReadOnlyCache } from '../../cache/DatabaseCaches'
+import { syncFunction } from '../../codeControl'
+import { RundownSyncFunctionPriority } from '../ingest/rundownInput'
+import { checkAccessAndGetPlaylist } from '../lib'
+import { CacheForStudio } from '../studio/cache'
+import { getStudioIdFromCacheOrLock, isStudioLock, StudioLock, studioLockFunction } from '../studio/syncFunction'
+import { CacheForPlayout, CacheForPlayoutPreInit } from './cache'
+
+export interface PlaylistLock extends StudioLock {
+	readonly _playlistId: RundownPlaylistId
+}
+
+export function isPlaylistLock(obj: any): obj is PlaylistLock {
+	const obj0 = obj as PlaylistLock
+	return isStudioLock(obj) && !!obj0._playlistId
+}
+
+export function getPlaylistIdFromCacheOrLock(cacheOrLock: any): RundownPlaylistId | undefined {
+	if (isPlaylistLock(cacheOrLock)) {
+		return cacheOrLock._playlistId
+	} else {
+		return (cacheOrLock as Partial<ReadOnlyCache<CacheForPlayoutPreInit>>).PlaylistId
+	}
+}
+
+/**
+ * Lock the playlist for performing a playout operation and load a cache of data.
+ * Note: This also locks the studio, if that is already locked then use 'rundownPlaylistPlayoutFromStudioLockFunction' instead
+ * @param context Meteor call context, to authenticate the request. Pass null if already authenticated
+ * @param contextStr Contextual information for the call to this function. to aid debugging
+ * @param rundownPlaylistId Id of the playlist to lock
+ * @param fcn Function to run while holding the lock
+ */
+export function playoutWithCacheLockFunction<T>(
```

Comment:
Just a minor suggestion: Could these functions be renamed into something more imperative? I don't know, perhaps `runWithPlayoutCacheLock` or something?